src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -301,6 +306,34 @@ def is_saveable_module(name, value):
 
             save_method(os.path.join(save_directory, pipeline_component_name), **save_kwargs)
 
+            if dduf_format:
+                import shutil
+                import tarfile
+
+                dduf_file_path = os.path.join(save_directory, dduf_filename)
+
+                if os.path.isdir(dduf_file_path):
+                    logger.warning(
+                        f"Removing the existing folder {dduf_file_path} so that we can save the DDUF archive."
+                    )
+                    shutil.rmtree(dduf_file_path)
+                if (
+                    os.path.exists(dduf_file_path)
+                    and os.path.isfile(dduf_file_path)
+                    and tarfile.is_tarfile(dduf_file_path)
+                ):
+                    # Open in append mode if the file exists
+                    mode = "a"
```

Comment:
Hmm, could this be problematic if we had the same file but it exists in a corrupted state?

Reply:
For this case, then we need to check before anything else if we do have an achieve and delete then. The logic for the append mode is to create the archive progressively as the models are getting saved. This way we use less disk memory

Reply:
Okay make sense but just wanted to flag this.